Expensify/App
# Fixed Confirm Password Validation in "Change Password" screen

@AndrewGable can you please review this?


### Details
- Added onBlur functions to compare the `newPassword` and `confirmPassword` fields. 
- `shouldShowPasswordConfirmError` flag to hide/show the validation message error.

### Fixed Issues

$ #4367 

### Tests
1.  Tested error scenario where both the passwords don't match
2. Tested scenario where both  the passwords match
3. Tested updating `newPassword` after updating `comparePassword`

### QA Steps

1. Enter different passwords for the newPassword field field and confirmPassword field.
2. Enter the confirm password first and then a new password to confirm the error is shown
3. Error should automatically go and the button should be enabled in case the passwords match
4. If the new password doesn't match the password rule, show the password hint
5. Disable button if old password is same as new password
6. Copy and paste the passwords instead of typing
7. Type an invalid old password, match both new and confirm passwords. Submitting should throw server error, then change either new or confirm password. Server error should hide and password mismatch error should be shown.

### Tested On


### Screenshots


#### Web
https://user-images.githubusercontent.com/3069065/128217828-ca3309bf-2e3e-42c6-998a-bf23c46d4068.mp4

#### Mobile Web
https://user-images.githubusercontent.com/3069065/128217957-cab5877f-d933-49a9-b9e9-68dca8f1f1e1.mp4

#### Desktop
https://user-images.githubusercontent.com/3069065/128218039-6a6c8fa0-4d6d-4148-a4da-cca4f0e6e328.mp4

#### iOS
https://user-images.githubusercontent.com/3069065/128220875-5a269402-4327-4b04-94d7-090843a59566.mp4

#### Android
https://user-images.githubusercontent.com/3069065/128220746-2ffdea76-9af6-419f-b23a-55c6864e0e30.mp4
